### ISSUE TITLE
Agent fails to install for Debian 12 / testing / bookworm

### ISSUE THREAD

**Issue by lod** [wazuh/wazuh-packages#2196]
|Wazuh version|Install type|Action performed|Platform|
|---|---|---|---|
| 4.4.1-1 | Agent | Install | Debian 12 |

After doing an apt install the following error is produced
```
Setting up wazuh-agent (4.4.1-1) ...
chmod: cannot access '/var/ossec/ruleset/sca/*': No such file or directory
dpkg: error processing package wazuh-agent (--configure):
```

The root cause of this is that the fallback debian sca.files list is empty.
https://github.com/wazuh/wazuh/blob/4.4/etc/templates/config/debian/sca.files

This causes no files to be placed in ruleset/sca/

So the wildcard fails to match anything and it all falls over.

It would be nice for there to be a cis file for Debian 12, but until that happens the install should not break.  The same fault will also happen on Debian unstable.

(As a side note, Ubuntu defaults to using the latest 22-04 cis file so there is always a rulset file to match against)

**Comment by rauldpm** [comment]
Hello @lod 

Debian 12 has not yet been officially released and therefore it is expected that certain files are missing, indeed, I confirm the error since I have been able to reproduce it in Debian 12, but not in Debian 11, which is expected.

- https://www.debian.org/releases/testing/releasenotes
  - This is a work-in-progress version of the Release Notes for Debian 12, codename bookworm, which isn't released yet. 
  - The information presented here might be inaccurate and outdated and is most likely incomplete.
  
```
The root cause of this is that the fallback debian sca.files list is empty.
https://github.com/wazuh/wazuh/blob/4.4/etc/templates/config/debian/sca.files
It would be nice for there to be a cis file for Debian 12, but until that happens the install should not break. The same fault will also happen on Debian unstable.
```

Indeed, I agree that an installation should not fail for this and that in the worst case, it should be installed without an SCA file, regarding the PR with the solution that you have proposed, I have made a review indicating the reasons why we cannot accept it right now.

- https://github.com/wazuh/wazuh-packages/pull/2197#discussion_r1188896483

Regarding having a generic SCA, I have been asking and mainly the reason seems to be to avoid false positives when applying an SCA that does not correspond to the same system as @72nomada has indicated to me.

Thus, we would have two ways, which solve the other:

1. Use a generic SCA

- If a generic SCA is used, it is not necessary to modify anything in the `wazuh-packages` repository since an SCA will always be applied

2.  Allow installations on systems without SCA

- This option allows us to install our software on systems that do not have an SCA, avoiding errors during the installation, the system would not have an SCA, but it could run certain functionalities.

---

Anyway, I encourage you to open an issue in the `wazuh/wazuh` repository with your request and your reasons: https://github.com/wazuh/wazuh/issues/new?assignees=&labels=&projects=&template=default.md&title=

**Comment by Oupsman** [comment]
Now that bookworm is released, do you have plans to allow the agent installation on a device running it ?

**Comment by DFolchA** [comment]
## Update

Fix problems when installing agents if the directory `/var/ossec/ruleset/sca` is empty.
https://github.com/wazuh/wazuh-packages/commit/9ea50f99e421ab77e3edb56dcee2b1a8401e7871

Test agent installation on Debian 12:

```
root@debian12:/home/vagrant# apt install ./wazuh-agent_4.5.0-1_amd64.deb 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'wazuh-agent' instead of './wazuh-agent_4.5.0-1_amd64.deb'
The following NEW packages will be installed:
  wazuh-agent
0 upgraded, 1 newly installed, 0 to remove and 3 not upgraded.
Need to get 0 B/8916 kB of archives.
After this operation, 30.2 MB of additional disk space will be used.
Get:1 /home/vagrant/wazuh-agent_4.5.0-1_amd64.deb wazuh-agent amd64 4.5.0-1 [8916 kB]
Preconfiguring packages ...
Selecting previously unselected package wazuh-agent.
(Reading database ... 49165 files and directories currently installed.)
Preparing to unpack .../wazuh-agent_4.5.0-1_amd64.deb ...
Unpacking wazuh-agent (4.5.0-1) ...
Setting up wazuh-agent (4.5.0-1) ...
root@debian12:/home/vagrant# ls /var/ossec/ruleset/sca/
root@debian12:/home/vagrant# 
```

Test manager installation on Debian 12:
```
root@debian12:/home/vagrant# apt install ./wazuh-manager_4.5.0-1_amd64.deb 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'wazuh-manager' instead of './wazuh-manager_4.5.0-1_amd64.deb'
Suggested packages:
  expect
The following NEW packages will be installed:
  wazuh-manager
0 upgraded, 1 newly installed, 0 to remove and 3 not upgraded.
Need to get 0 B/171 MB of archives.
After this operation, 631 MB of additional disk space will be used.
Get:1 /home/vagrant/wazuh-manager_4.5.0-1_amd64.deb wazuh-manager amd64 4.5.0-1 [171 MB]
Selecting previously unselected package wazuh-manager.
(Reading database ... 49165 files and directories currently installed.)
Preparing to unpack .../wazuh-manager_4.5.0-1_amd64.deb ...
Unpacking wazuh-manager (4.5.0-1) ...
Setting up wazuh-manager (4.5.0-1) ...
root@debian12:/home/vagrant# ls /var/ossec/ruleset/sca/
cis_amazon_linux_1.yml.disabled     cis_mongodb_36.yml.disabled		  cis_sqlserver_2016.yml.disabled
cis_amazon_linux_2.yml.disabled     cis_mysql5-6_community.yml.disabled   cis_sqlserver_2017.yml.disabled
cis_apache_24.yml.disabled	    cis_mysql5-6_enterprise.yml.disabled  cis_sqlserver_2019.yml.disabled
cis_apple_macOS_10.11.yml.disabled  cis_nginx_1.yml.disabled		  cis_ubuntu14-04.yml.disabled
cis_apple_macOS_10.12.yml.disabled  cis_oracle_database_19c.yml.disabled  cis_ubuntu16-04.yml.disabled
cis_apple_macOS_10.13.yml.disabled  cis_postgre-sql-13.yml.disabled	  cis_ubuntu18-04.yml.disabled
cis_apple_macOS_10.14.yml.disabled  cis_rhel5_linux.yml.disabled	  cis_ubuntu20-04.yml.disabled
cis_apple_macOS_10.15.yml.disabled  cis_rhel6_linux.yml.disabled	  cis_ubuntu22-04.yml.disabled
cis_apple_macOS_11.1.yml.disabled   cis_rhel7_linux.yml.disabled	  cis_win10_enterprise.yml.disabled
cis_apple_macOS_12.0.yml.disabled   cis_rhel8_linux.yml.disabled	  cis_win11_enterprise.yml.disabled
cis_centos6_linux.yml.disabled	    cis_rhel9_linux.yml.disabled	  cis_win2012r2.yml.disabled
cis_centos7_linux.yml.disabled	    cis_sles11_linux.yml.disabled	  cis_win2016.yml.disabled
cis_centos8_linux.yml.disabled	    cis_sles12_linux.yml.disabled	  cis_win2019.yml.disabled
cis_debian10.yml.disabled	    cis_sles15_linux.yml.disabled	  cis_win2022.yml.disabled
cis_debian7.yml.disabled	    cis_solaris11.4.yml.disabled	  sca_unix_audit.yml.disabled
cis_debian8.yml.disabled	    cis_solaris11.yml.disabled		  web_vulnerabilities.yml.disabled
cis_debian9.yml.disabled	    cis_sqlserver_2012.yml.disabled
cis_iis_10.yml.disabled		    cis_sqlserver_2014.yml.disabled
```

Test agent upgrade:
```
root@debian12:/home/vagrant# apt install ./wazuh-agent_4.5.0-1_amd64.deb 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'wazuh-agent' instead of './wazuh-agent_4.5.0-1_amd64.deb'
The following packages will be upgraded:
  wazuh-agent
1 upgraded, 0 newly installed, 0 to remove and 3 not upgraded.
1 not fully installed or removed.
Need to get 0 B/8916 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 /home/vagrant/wazuh-agent_4.5.0-1_amd64.deb wazuh-agent amd64 4.5.0-1 [8916 kB]
Reading changelogs... Done
Preconfiguring packages ...
(Reading database ... 49537 files and directories currently installed.)
Preparing to unpack .../wazuh-agent_4.5.0-1_amd64.deb ...
Unpacking wazuh-agent (4.5.0-1) over (4.4.4-1) ...
Setting up wazuh-agent (4.5.0-1) ...
```
Test installation and upgrade for the following systems:
- [x] Debian 12
- [x]  Debian 11
- [x] Debian 10
- [x] Ubuntu 18
- [x] Ubuntu 20
- [x] Ubuntu 22

**Comment by jnasselle** [comment]
Closed by #2256 and #2196
